api/hypershift/v1alpha1/nodepool_types.go

```diff
@@ -209,6 +209,10 @@ type NodePoolSpec struct {
 	// +kubebuilder:validation:Enum=arm64;amd64
 	// +optional
 	Arch string `json:"arch,omitempty"`
+
+	// FailureDomain is the preferred failure domain for the NodePool.
+	// +optional
+	FailureDomain *string `json:"failureDomain,omitempty"`
```

Comment:
Adding this to v1alpha1 as well as v1beta1 doesn't look right to me. Is this normally done in HyperShift?

Reply:
I asked myself this and looking at git history, this seems common practise here.

Reply:
If there's no conversion webhook then this makes sense, I don't know if they have one or not